server/service/src/sync/file_sync_driver.rs

```diff
@@ -0,0 +1,196 @@
+use std::sync::Arc;
+
+use crate::{
+    service_provider::ServiceProvider, settings::Settings, static_files::StaticFileService,
+    sync::file_synchroniser::FileSynchroniser,
+};
+
+use super::settings::SyncSettings;
+use tokio::{
+    sync::mpsc::{self, Receiver, Sender},
+    time::Duration,
+};
+use util::is_central_server;
+
+const FILE_SYNC_UPLOAD_DELAY: Duration = Duration::from_millis(100); // This just gives time for a PAUSE message to be received between uploading files
+const FILE_SYNC_NO_FILES_DELAY: Duration = Duration::from_millis(10000); // If there's nothing to upload or there was an error, wait a longer before checking again
+
+pub enum FileSyncMessage {
+    Start,   // Start sync (could be manual trigger, or automatic on server startup)
+    Stop,    // Stop sync (could be manual trigger, or automatic on server shutdown)
+    Pause, // Pause sync this is called by the main sync process to pause the file sync during a normal sync operation
+    UnPause, // Restart sync if it's not Stopped
+}
+
+pub struct FileSyncDriver {
+    receiver: Receiver<FileSyncMessage>,
+    static_file_service: Arc<StaticFileService>,
+}
+
+#[derive(Clone)]
+pub struct FileSyncTrigger {
+    sender: Sender<FileSyncMessage>,
+}
+
+/// Used to 'drive' file sync synchronisation, it's tasks:
+/// * Expose channel for manually triggering sync
+/// * Trigger sync every SyncSettings.interval_seconds (only when initialised)
+impl FileSyncDriver {
+    pub fn init(settings: &Settings) -> (FileSyncTrigger, FileSyncDriver) {
+        // We use a single-element channel so that we can only have one trigger message processed at a time (no need to queue them up?)
```

Comment:
I wonder if this could somehow cause us problems with slow connections, long integrations or something?

Reply:
Yeah we kind of want to consume only the latest message, but not sure what's the a nice way to do that.